### PR TITLE
fix apple clang compiler warning

### LINKED ISSUE #6102: Build fail on x86 mac
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
```
cmake  -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_C_COMPILER=clang -DCMAKE_BUILD_TYPE=Debug ..
export CXXFLAGS="-w"
export CFLAGS="-w"
make gtests_dbms
```
Prints out error
```
clang: warning: argument unused during compilation: '-no-pie' [-Wunused-command-line-argument]
clang: warning: argument unused during compilation: '-no-pie' [-Wunused-command-line-argument]
[ 18%] Linking C static library libxxhashd.a
[ 18%] Building C object contrib/lz4-cmake/CMakeFiles/lz4.dir/__/lz4/lib/lz4hc.c.o
clang: warning: argument unused during compilation: '-no-pie' [-Wunused-command-line-argument]
[ 18%] Linking C static library liblz4d.a
clang: error: argument unused during compilation: '-no-pie' [-Werror,-Wunused-command-line-argument]
make[3]: *** [libs/libcommon/CMakeFiles/apple_rt.dir/src/apple_rt.cpp.o] Error 1
make[2]: *** [libs/libcommon/CMakeFiles/apple_rt.dir/all] Error 2
make[2]: *** Waiting for unfinished jobs....
[ 18%] Built target xxhash
[ 18%] Built target lz4
clang: error: argument unused during compilation: '-no-pie' [-Werror,-Wunused-command-line-argument]
```

### 4. What is your TiFlash version? (Required)
master

### PR DESCRIPTION
**Pull request by Lloyd-Pottiger**
Signed-off-by: Lloyd-Pottiger <yan1579196623@gmail.com>

### What problem does this PR solve?

Issue Number: close #6102 

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -39,7 +40,7 @@ if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
         message (FATAL_ERROR "GCC version must be at least 7! For example, if GCC 7 is available under gcc-7, g++-7 names, do the following: export CC=gcc-7 CXX=g++-7; rm -rf CMakeCache.txt CMakeFiles; and re run cmake or ./release.")
     endif ()
     set(MOST_DEBUGGABLE_LEVEL -Og)
-elseif (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+elseif (CMAKE_CXX_COMPILER_ID MATCHES "AppleClang" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
```
Comment (JaySon-Huang) [inline CMakeLists.txt:43]:
```suggestion
elseif (CMAKE_CXX_COMPILER_ID MATCHES "AppleClang|Clang")

```

File: cmake/test_compiler.cmake [hunk 2]
```diff
@@ -17,7 +17,7 @@ include (CMakePushCheckState)
 
 cmake_push_check_state ()
 
-if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+if (CMAKE_CXX_COMPILER_ID MATCHES "AppleClang" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
```
Comment (JaySon-Huang) [inline cmake/test_compiler.cmake:20]:
```suggestion
if (CMAKE_CXX_COMPILER_ID MATCHES "AppleClang|Clang")
```

File: CMakeLists.txt (hunk 1 above)
Comment (JaySon-Huang) [reply CMakeLists.txt:43]: Use FATAL_ERROR in line 50 when meet unknown compiler

File: cmake/test_compiler.cmake (hunk 2 above)
Comment (Lloyd-Pottiger) [reply cmake/test_compiler.cmake]: thanks, fixed!

**Review by JaySon-Huang — APPROVED** [review]

**Review by CalvinNeo — APPROVED** [review]
lgtm

**Pull request merged**